### ISSUE TITLE
Bug: typescript-eslint 6.3.0 and clsx

### ISSUE THREAD

**Issue by michalbundyra** [typescript-eslint/typescript-eslint#7438]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

eslint-plugin-tslint

### Repro Code

```typescript
import { clsx } from 'clsx'
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

no errors

### Actual Result

`Parse errors in imported module 'clsx': Cannot find module 'tsutils/util/util'`

### Additional Info

The issue is present since 6.3.0 release. All good on previous version.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.3.0` |
| `@typescript-eslint/parser`            | `6.3.0` |
| `@typescript-eslint/rule-tester`       | `-` |
| `@typescript-eslint/scope-manager`     | `-` |
| `@typescript-eslint/typescript-estree` | `-` |
| `@typescript-eslint/type-utils`        | `-`|
| `@typescript-eslint/utils`             | `-` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.46.0` |
| `node`                                 | `18.12.0` |

**Comment by JoshuaKGoldberg** [comment]
> **Playground Link**
> No response

Please fill out the issue template fully when filing an issue. If we can't reproduce your issue locally just from the instructions there, we can't help you. And I can't reproduce this issue locally.

Looks like you probably want [🐛 Report a Complex Bug With a Reproduction Repository](https://github.com/typescript-eslint/typescript-eslint/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=08-bug-report-complex.yaml&title=Bug%3A+%3Cshort+description+of+the+issue%3E).
